src/cpp11.cpp

```diff
@@ -4,6 +4,7 @@
 #include "vroom_types.h"
 #include "cpp11/declarations.hpp"
 #include <R_ext/Visibility.h>
+#include <cstddef>
```

Comment:
I think if we were going to put this anywhere, it would be in `vroom_types.h`, since the idea of that file is that it gets automatically included here to provide the definitions for any types needed to support this autogenerated file, like `vroom_errors`
https://github.com/tidyverse/vroom/blob/27285a6e6d3e8365dea79b2c1e3d019b8b088cd7/src/cpp11.cpp#L87

But I'm not super worried about this one because `cpp11/declarations.hpp` includes `cpp11/cpp11.hpp` which includes all of cpp11, and `cpp11/r_vector.hpp` includes `<stddef.h>` which is where `ptrdiff_t` is (https://cplusplus.com/reference/cstddef/). Typically I am more worried about implicit dependencies like this, but this file is kind of an exception.

---

AFAICT, it was very intentional that `cpp11/declarations.hpp` includes `cpp11/cpp11.hpp`. It basically includes that file + adds a few other declarations needed for `code.cpp`. But the idea is that including `cpp11/declarations.hpp` is "enough" to generate `code.cpp` without any other cpp11 headers. So we don't need to add `cpp11/as.hpp` or `cpp11/integers.hpp`
https://github.com/r-lib/cpp11/blob/500f642b4ea132ec8c168fc70a28e81e9510ece3/inst/include/cpp11/declarations.hpp#L7-L16

Reply:
I think if CRAN starts enforcing on this clang finding, it's gonna be chaos for sure.